packages/jest-preset-default/scripts/babel-transformer.js

```diff
@@ -0,0 +1,27 @@
+const fs = require( 'fs' );
+const babelJest = require( 'babel-jest' );
+const babelJestTransformer = babelJest.createTransformer();
+
+module.exports = {
+	...babelJestTransformer,
+	getCacheKey( src, filename, ...args ) {
+		const isBlockIndex = /block-library[\/\\]src[\/\\].+[\/\\]index\.js/.test( filename );
```

Comment:
I don't think it's very useful as part of `@wordpress/jest-preset-default` in the current form. The issue is that this check is very specific to the structure of the Gutenberg repository and it will never match with any block definition which is maintained by other projects.

Moving forward, I see 2 ways of handling it. We either move this override to https://github.com/WordPress/gutenberg/blob/master/test/unit/jest.config.js which is local to Gutenberg or we find a way to approach it for all projects which load `block.json`. Given that we want to rework how `block.json` is loaded as proposed in #16088, it might be better to apply this transform to local unit tests config.

Reply:
Thanks for the feedback. Yep, you're right, it shouldn't be in that package. I've moved it to the appropriate place. It still seems to work.